test/amqp-test.js

```diff
@@ -0,0 +1,154 @@
+      assert(channel.assertQueue.calledBefore(channel.bindQueue))
+
+      assert(channel.assertQueue.calledOnce)
+      assert(channel.assertQueue.calledWithExactly(queueName))
+
+      assert(channel.bindQueue.calledOnce)
+      assert(channel.bindQueue.calledWithExactly(queueName, exchangeName, routingKey))
+    })
+
+    it('should return channel', async () => {
+      const channel = new ChannelMock()
+
+      const actual = await assertChannel(randomConfig(), channel)
+
+      assert.strictEqual(channel, actual)
+    })
+  })
+
+  // describe('initChannel', () => {
+  //   it('should create channel', () => {
+  //     const expected = {}
+  //     const actual = initChannel(() => expected, () => {}, randomConfig())
+  //
+  //     assert.strictEqual(expected, actual)
+  //   })
+  //
+  //   it('should assert channel', () => {
+  //     const assertChannel = spy((config, channel) => channel)
+  //     const config = randomConfig()
+  //     const channel = {}
+  //
+  //     const actual = initChannel(setup => setup(channel), assertChannel, config)
+  //
+  //     assert.strictEqual(channel, actual)
+  //     assert(assertChannel.calledOnce)
+  //     assert(assertChannel.calledWithExactly(config, channel))
+  //   })
+  // })
+
+  describe('consumeFrom', () => {
+    it('should consume from correct queue on the channel', async () => {
+      const channel = new ChannelMock({})
+      const queueName = randomString()
+
+      const consume = consumeFrom(() => channel)
+      await consume({ queueName }, () => {})
+
+      calledOnceWithArgs(channel.consume, is(queueName), isFunction)
+    })
+
+    it('should call ack only when acking', async () => {
+      const message = {}
+      const channel = new ChannelMock(message)
+      const consume = consumeFrom(() => channel)
+
+      const handler = spy((ack, nack, msg) => ack(msg))
+      await consume('', handler)
+
+      calledOnceWithArgs(handler, isFunction, isFunction, is(message))
+      assert(channel.ack.calledWithExactly(message))
+      assert.strictEqual(0, channel.nack.callCount)
+    })
+
+    it('should call nack only when nacking', async () => {
+      const message = {}
+      const channel = new ChannelMock(message)
+      const consume = consumeFrom(() => channel)
+
+      const handler = spy((ack, nack, msg) => nack(msg))
+      await consume('', handler)
+
+      calledOnceWithArgs(handler, isFunction, isFunction, is(message))
+      assert(channel.nack.calledWithExactly(message))
+      assert.strictEqual(0, channel.ack.callCount)
+    })
+  })
+
+  describe('publishTo', () => {
+    it('should return a publisher function', () => {
+      const publish = publishTo(() => {})(randomConfig())
+      assert.strictEqual(typeof publish, 'function')
+    })
+
+    it('should publish to channel', async () => {
+      const channel = new ChannelMock()
+      const config = randomConfig()
+      const { exchangeName, routingKey } = config
```

Comment:
I see this only uses exchangeName & routingKey, like I'd expect but your ChannelMock still asserts queue. So just looking for clarity.